sdk/Trace/ExporterFactory.php

```diff
@@ -0,0 +1,152 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Sdk\Trace;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Psr7\HttpFactory;
+use Nyholm\Dsn\DsnParser;
+use OpenTelemetry\Contrib\Jaeger\Exporter as JaegerExporter;
+use OpenTelemetry\Contrib\Newrelic\Exporter as NewrelicExporter;
+use OpenTelemetry\Contrib\Otlp\Exporter as OtlpExporter;
+use OpenTelemetry\Contrib\OtlpGrpc\Exporter as OtlpGrpcExporter;
+use OpenTelemetry\Contrib\Zipkin\Exporter as ZipkinExporter;
+use OpenTelemetry\Contrib\ZipkinToNewrelic\Exporter as ZipkinToNewrelicExporter;
+
+class ExporterFactory
+{
+    private $name;
+    private $_allowedExporters = ['jaeger' => true, 'zipkin' => true, 'newrelic' => true, 'otlp' => true, 'otlpgrpc' => true, 'zipkintonewrelic' => true];
+
+    public function __construct(string $name)
+    {
+        $this->name = $name;
+    }
+
+    /**
+      * Returns the coresponding Exporter via the configuration string
+      *
+      * @param string $configurationString String containing unextracted information for Exporter creation
+      * Should follow the format: contribType+baseUrl?option1=a
+      * Query string is optional and based on the Exporter
+      */
+    public function fromConnectionString(string $configurationString)
+    {
+        $strArr = explode('+', $configurationString);
+        // checks if input is given with the format type+baseUrl
+        if (sizeof($strArr) != 2) {
+            return null;
+        }
+
+        $contribName = strtolower($strArr[0]);
+        $endpointUrl = $strArr[1];
+
+        if (!$this->_isAllowed($contribName)) {
+            return null;
+        }
+
+        // endpointUrl should only be null with otlp and otlpgrpc
+        $licenseKey = '';
+        if ($endpointUrl != false) {
+            $dsn = DsnParser::parse($endpointUrl);
+            $endpointUrl = (string) ($dsn->withoutParameter('licenseKey'));
+            $licenseKey = (string) ($dsn->getParameter('licenseKey'));
+        }
```

Comment:
This is overly-specific.  The handling of `licenseKey` seems to only be applicable to the New Relic-flavored exporters.  When new exporters are added that may have their own options this handling logic will need to change.  Instead, if URL query string parameters are to be the mechanism for conveying configuration to exporters then they should be pulled into an array that is passed to the relevant constructor.